core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -525,7 +530,38 @@ private static SqlNode groupItem(List<SqlNode> groupKeys,
   /** Visits a TableScan; called by {@link #dispatch} via reflection. */
   public Result visit(TableScan e) {
     final SqlIdentifier identifier = getSqlTargetTable(e);
-    return result(identifier, ImmutableList.of(Clause.FROM), e, null);
+    SqlNode node = identifier;
+    if (e instanceof Hintable) {
+      final ImmutableList<RelHint> hints = e.getHints();
+      if (hints != null && ! hints.isEmpty()) {
+        SqlParserPos pos = identifier.getParserPosition();
```

Comment:
Does the `Hintable.getHints` has any chance to return null ?

Reply:
No this seems not to be the case. I will remove this check.